From 03f02ce1a7e34e0706e9fb3a5d70311418beba8a Mon Sep 17 00:00:00 2001
From: Alex Leibovich <alexl@marvell.com>
Date: Mon, 6 May 2019 14:07:45 +0300
Subject: [PATCH 1110/1200] defconfig: mvebu: add configurable boot device and
 partition

Adding ability to configure boot device and
partition with configuration file.

Change-Id: I95f8a52bd2f1d4fddce23d15e092f5559ffcca60
Signed-off-by: Alex Leibovich <alexl@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/8629
Reviewed-by: Kostya Porotchkin <Kostya.Porotchkin@cavium.com>
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
---
 cmd/mvebu/Kconfig                     | 15 +++++++++++++++
 include/configs/mvebu_armada-common.h | 14 +++++---------
 2 files changed, 20 insertions(+), 9 deletions(-)

diff --git a/cmd/mvebu/Kconfig b/cmd/mvebu/Kconfig
index 89f42cf39b..4524631bb7 100644
--- a/cmd/mvebu/Kconfig
+++ b/cmd/mvebu/Kconfig
@@ -78,6 +78,21 @@ config MVEBU_MMC_BOOT
 
 endchoice
 
+config MVEBU_BOOT_PART
+	int "boot partition"
+	default 0
+	help
+	  sets boot partition, and sets a specific
+	  partition as a target for "bubt" command.
+	  For details about bubt command please see the documentation
+	  in doc/mvebu/cmd/bubt.txt
+
+config MVEBU_BOOT_DEVICE
+	int "boot device id"
+	default 0
+	help
+	  Used for setting boot device id.
+
 config MVEBU_UBOOT_DFLT_NAME
 	string "Default image name for bubt command"
 	default "flash-image.bin"
diff --git a/include/configs/mvebu_armada-common.h b/include/configs/mvebu_armada-common.h
index b40cfa42d4..382fae74fe 100644
--- a/include/configs/mvebu_armada-common.h
+++ b/include/configs/mvebu_armada-common.h
@@ -144,15 +144,11 @@ defined(CONFIG_TARGET_MVEBU_ARMADA_37XX) && defined(CONFIG_MV88E6XXX_SWITCH)
 #endif
 #endif
 
-#ifdef CONFIG_ENV_IS_IN_MMC
-#ifdef CONFIG_TARGET_MVEBU_ARMADA_37XX
-#define CONFIG_SYS_MMC_ENV_DEV		1 /* A3700 emmc is on sdhci-1*/
-#define CONFIG_SYS_MMC_ENV_PART		1 /* 0-DATA, 1-BOOT0, 2-BOOT1 */
-#else
-#define CONFIG_SYS_MMC_ENV_DEV		0
-#define CONFIG_SYS_MMC_ENV_PART		0 /* 0-DATA, 1-BOOT0, 2-BOOT1 */
-#endif
-#endif
+/*
+ * Device ID and Boot partition defined in external script
+ */
+#define CONFIG_SYS_MMC_ENV_DEV		(CONFIG_MVEBU_BOOT_DEVICE)
+#define CONFIG_SYS_MMC_ENV_PART		(CONFIG_MVEBU_BOOT_PART)
 
 /*
  * SATA/SCSI/AHCI configuration
-- 
2.22.0

